sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -589,6 +590,21 @@ int gnrc_ndp_internal_sl2a_opt_handle(gnrc_pktsnip_t *pkt, ipv6_hdr_t *ipv6, uin
         return -EINVAL;
     }
 
+#ifdef MODULE_GNRC_SIXLOWPAN_ND
+    gnrc_ipv6_netif_t *ipv6_iface = gnrc_ipv6_netif_get(iface);
+
+    if (ipv6_iface->flags & GNRC_IPV6_NETIF_FLAGS_SIXLOWPAN) {
+        // https://tools.ietf.org/html/rfc4944#section-8
+        if (sl2a_opt->len == 1) {
+            sl2a_len = 2;
```

Comment:
Since we have also 1-bit devices in mind to use with 6Lo, this might not be true